### DISCUSSION TITLE
Requesting for smooth scrolling and page number shortcut

### DISCUSSION THREAD

**Discussion by Miraj2407** [sumatrapdfreader/sumatrapdf#3479] [Ideas]
Please, add _smooth scrolling_ like the " **PDF Reader by XODO** " Application. 
And add _page number shortcut_ - this feature exists on the _previous version_ but is **missing** in the present version.

**Comment by GitHubRulesOK** [comment]
there is as much attempts at scrolling as possible but its down to hardware in most cases as rendering lines of one line at a time is natually stepped. Using my mouse or touchpad they are about as smooth as Acrobat Edge

what shortcut AFAIK the page numbers are still shown but hotkey is shift `I` (lowercase is now **i**nvert) in current pre-release.
you can swap them back over, if you use advanced settings shortcut reassignments.
![image](https://user-images.githubusercontent.com/12754913/236246915-ee87adbc-b901-4ccc-b216-12ed4fcc5e92.png)

**Reply by Miraj2407** [reply]
### Thanks for your second suggestion this helps me to rediscover my favorite feature.

But in the first case- Your opinion will somehow be correct. You compared this Softower with  **Acrobat Edge** but I suggested you to use **PDF Reader by XODO** " Application in this matter for a better understanding of my message. Because the two software is currently running on my same configuration laptop, so I can differentiate the scrolling feelings. _It's my humble request to check **PDF Reader by XODO** before replying._

**Reply by GitHubRulesOK** [reply]
Cant find a portable Xodo (my policy is no 3rd party ms or google store downloads) Last time I did download it was deleted as invasive. 

Thus depends what your using.
So using the Xodo online (canvas viewer) its 4 clicks to move down an inch, but in SumatraPDF for **same document and zoom** its about 11 clicks. So SumatraPDF is more than twice (almost 3 times) as smooth using this mouse. OR a different scale of touchpad stepping or using Auto scroll in SumatraPDF its many times smoother (not available in my Xodo viewer)

Whole page  is 4 strokes of the wheel in Xodo but 14 strokes in SumatraPDF thus again its slower and smoother to scroll in SumatraPDF.
As said its down to hardware not app as to what input is sent to a viewer, and a fine balance as to how to jerk so many lines of text for that incoming instruction. its done based on zoomed pixels at time of stepped keyboard/mouse/pad instructions. Half the number of pixels and slower speed will look smoother.

**Reply by Miraj2407** [reply]
**Without walking my road you can not feel my problem...**

**Reply by GitHubRulesOK** [reply]
dugout the portable copy from last year (still not portable insists on installing tons of this and that into windows system folders on more than one drive!! Talk about avoid bloatware its in a class of its own (possibly worse than Acrobat).

Any way still could not see any scroll based improvement over SumatraPDF, still looked more jerky. 
And security seems to have stopped it from working any more, so again will try to find what to delete from system folders.

**Reply by Miraj2407** [reply]
OK 😶

**Reply by Miraj2407** [reply]
### Is SumatraPDF reader has Google Chrome pdf reader's feature _slide show_? If not Please add


![Screenshot (1025)](https://user-images.githubusercontent.com/132584906/236541385-2589f4e5-8ac6-4c92-b58a-81c7d97e7096.png)

**Comment by KillianLobato** [comment]
Hi everyone,
I can confirm that Xodo scrolling is extremely smooth, when using a touchpad. It scrolls just like macOS Preview software. We are not about talking about smooth animation from one wheel scroll click to the next. That is smooth animation. We are talking about pixel-by-pixel scrolling when the input is the trackpad or some microsoft mice.
Also, if you use two fingers on the trackpad, Xodo allows you to diagonaly scroll the page. In Sumatra you can either scroll up/down or left/right.
I think care must be taken when taking inputs from a trackpad and from a mouse. The way one interacts with the PDF is different.
See my other post from January of this year where I started discussing how Sumatra could be developed to improve user interaction with it.
Kind regards,
KL
